modules/solid_mechanics/src/actions/DomainIntegralAction.C

```diff
@@ -373,11 +388,11 @@ DomainIntegralAction::act()
     const std::string uo_type_name("CrackFrontDefinition");
 
     InputParameters params = _factory.getValidParams(uo_type_name);
-    if (_use_crack_front_points_provider && _used_by_xfem_to_grow_crack)
+    if (_use_crack_front_points_provider && !xfem_exec_flags.empty())
     {
-      params.set<ExecFlagEnum>("execute_on") = {EXEC_INITIAL, EXEC_TIMESTEP_END, EXEC_NONLINEAR};
       // The CrackFrontDefinition updates the vpps and MUST execute before them
       params.set<int>("execution_order_group") = -1;
+      params.set<ExecFlagEnum>("execute_on") = xfem_exec_flags;
```

Comment:
I think we need `EXEC_INITIAL` here, and that's not in `xfem_exec_flags`. I'd like to talk about the lingering issues noted in the TODO above too.

Reply:
This is trying handle the order of execution for the vpps and user_objects.  We wanted to handle this in a separate PR.  @dschwen was going to help us get these straightened out in a separate PR.

Reply:
Yeah, I get that. I just see that `EXEC_INITIAL` was in the list of exec flags, but it's not in the list defined on line 378. I think maybe the fix is to just add `EXEC_INITIAL` to `xfem_exec_flags` on line 378. Apparently we don't have tests that catch it, but I do recall that this was important.